src/App.js

```diff
@@ -1,16 +1,87 @@
-import React from 'react';
+import React from "react";
+import ToDoList from "./components/TodoComponents/TodoList";
+import ToDoForm from "./components/TodoComponents/TodoForm";
+import { ifStatement } from "@babel/types";
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
-  render() {
-    return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
-      </div>
-    );
-  }
+	// you will need a place to store your state in this component.
+	// design `App` to be the parent component of your application.
+	// this component is going to take care of state, and any change handlers you need to work with your state
+	constructor() {
+		super();
+		this.state = {
+			toDosArray: [
+				{
+					task: "Organize Garage",
+					id: 1528817077286,
+					completed: false
+				},
+				{
+					task: "Bake Cookies",
+					id: 1528817084358,
+					completed: false
+				}
+			],
+			newTodoField: ""
+		};
+	}
+
+	formInputChange = e => {
+		this.setState({
+			newTodoField: e.target.value
+		});
+	};
+
+	addToDo = e => {
+		e.preventDefault();
+		const newTodoArrayItem = {
+			task: this.state.newTodoField,
+			completed: false,
+			id: Date.now()
+		};
+		this.setState({
+			toDosArray: [...this.state.toDosArray, newTodoArrayItem],
+			newTodoField: ""
+		});
+	};
+
+	selectExistingToDo = e => {
+		const selected = [...this.state.toDosArray].map(el => {
```

Comment:
I know I was all over the place trying to work through this with you yesterday. Looking over it, I'm not sure we have to copy the array before we map over it, since we know map won't mutate the array. Like I said though, sometimes I will copy it just in case though, besides like map I have to double check if an array method will mutate the original. If someone were to come through here and change the logic to use a different array method, they'd have that safety buffer of the array being copied already.